### ISSUE TITLE
screenReaderMode should become a renderer addon

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#2095]
After https://github.com/xtermjs/xterm.js/issues/2005 is done it would make a lot of sense and clean up a bunch of code if we changed `screenReaderMode` to just be another renderer type.

**Comment by Tyriar** [comment]
I changed my mind here. While they are very similar, it should work regardless of renderer. It would also duplicate a bunch of domrenderer code for no reason.
